src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1032,7 +1033,8 @@ public SocketError ReceiveFromAsync(IList<ArraySegment<byte>> buffers, SocketFla
             }
         }
 
-        public SocketError ReceiveMessageFrom(byte[] buffer, int offset, int count, ref SocketFlags flags, byte[] socketAddress, ref int socketAddressLen, bool isIPv4, bool isIPv6, int timeout, out IPPacketInformation ipPacketInformation, out int bytesReceived)
+        public SocketError ReceiveMessageFrom(
```

Comment:
Yeah, now that you point it out, I can't see why we have the duplicated code in the first place.

I was going to say "it saves us an if() somewhere down in TryCompleteReceiveFrom (etc)", but even that doesn't actually seem to be the case.

So I agree, there's no point in duplicating the code here.  I'll file an issue to clean up the other cases as well.